### ISSUE TITLE
French typography : angle quotes follower/preceded by non-breakable space

### ISSUE THREAD

**Issue by borisschapira** [markdown-it/markdown-it#115]
In French, double quotes are represented by angle quotes, or "guillemets", separated by non-breaking spaces « like this ».
Is there a way to define the "quotes" parameter to integrate those configuration.

For example, instead of 

```
quotes: '“”‘’'
```

We could have :

```
quotes: 
 - '“'
 - '”'
 - '‘'
 - '’'
```

And, in French :

```
quotes: 
 - '«&nbsp;'
 - '&nbsp;»'
 - '“'
 - '”'
```

What do you think ?

**Comment by puzrin** [comment]
We could add alternate option format, as object, with fields `doubleOpen`, `doubleClose`, `singleOpen`, `singleClose`.

**Comment by borisschapira** [comment]
@puzrin I don't know if it's necessary. I've tried what I have proposed and it "nearly" worked...
Maybe a simple fix could help, without the need to complexify the configuration with new keywords...

**Comment by puzrin** [comment]
You are right, passing array looks good.

**Comment by borisschapira** [comment]
How did you configured it to make it work ?

**Comment by puzrin** [comment]
It doesn't work yet

**Comment by rlidwka** [comment]
Fixed. Now you can supply an array there, and it should work as expected.

**Comment by borisschapira** [comment]
@rlidwka thanks, seems to work with multiple caracters like the one you've set in the test example, but I can't manage to make it work with special chars like `['«\xA0', '\xA0»', '‹\xA0', '\xA0›']`, I only comes from : `"foo" 'bar'` to `«\xA0foo\xA0» ‹\xA0bar\xA0›`. Shouldn't the `\xA0` be converted to an `&nbsp;` ?

**Comment by puzrin** [comment]
@borisschapira could you explain what's wrong? `0xA0` is &nbsp; code and it's a valid output.

**Comment by borisschapira** [comment]
I'm trying to use it for a static generator, [Hexo](https://github.com/hexojs/hexo). So far, the `\xA0` is rendered without alteration in my HTML, producing

`«xA0foo\xA0» ‹\xA0bar\xA0›` 

instead of ;

`« foo » ‹ bar ›`

**Comment by borisschapira** [comment]
Example content : http://next.borisschapira.com/2015/06/03/test-typo/

**Comment by rlidwka** [comment]
Here is the test I'm using:

```js
var markdownit = require('markdown-it')

var md = markdownit({
  typographer: true,
  quotes: [ '«\xA0', '\xA0»', '‘', '’' ]
})

console.log(md.render('"foo"'))
// <p>« foo »</p>

console.log(Buffer(md.render('"foo"')))
// <Buffer 3c 70 3e c2 ab c2 a0 66 6f 6f c2 a0 c2 bb 3c 2f 70 3e 0a>
//                        ^^^^^          ^^^^^
```

Added hex just to make sure the right sequence (`c2 a0` in utf-8) is produced.

Maybe Hexo is replacing non-printable characters with their char codes for whatever reason?

**Comment by borisschapira** [comment]
You're right, my issue is between the configuration of markdown-it and it's usage : '\' are escaped, so that my code is, in fine :

```js
var md = markdownit({
  typographer: true,
  quotes: options.quotes // [ '«\\xA0', '\\xA0»', '‘', '’' ]
})
```

I'll look into it, thanks.

**Comment by borisschapira** [comment]
FYI, found the issue (or what I think is this issue, but maybe I do not know the YAML standard enough) in [js-yaml](https://github.com/nodeca/js-yaml/issues/193).

**Comment by borisschapira** [comment]
Do you plan on releasing a new version in npm ?

**Comment by vyp** [comment]
@borisschapira the fix is in 4.2.2?

**Comment by borisschapira** [comment]
@vyp yes it is, I used it alright.
